### PR TITLE
Issue#8 : Adding security for admins (excluding GET).

### PR DESCRIPTION
**Pull request by kunalkhosla**
TODO: Get admin list from the repository, and remove hardcoded dummy admin

### CODE REVIEW

**Comment by varunmehta** [comment]
The hardcoded dummy admin is created by spring-security/boot. I would want to leave it for some time as of now.
http://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-security.html 

We blow up the DB after a few events, as we are restricted to 10K records across the DB (all tables)

> The default AuthenticationManager has a single user (‘user’ username and random password, printed at INFO level when the application starts up)
> // This is a just an example.
> Using default security password: 78fa095d-3f4c-48b1-ad50-e24c31d5cf35
> 
> If you fine-tune your logging configuration, ensure that the org.springframework.boot.autoconfigure.security category is set to log INFO messages, otherwise the default password will not be printed.
>
> You can change the password by providing a security.user.password. This and other useful properties are externalized via SecurityProperties (properties prefix "security").

File: src/main/java/org/glenrockindiancommunity/WebSecurityConfig.java [hunk 1]
```diff
@@ -11,33 +12,31 @@
...
+        http.httpBasic().and().authorizeRequests()
+                .antMatchers(HttpMethod.GET,"/event/**").permitAll()
+                .antMatchers("/event/**")
+                .hasRole("ADMIN")
+                .and()
+                .antMatcher("/**")
+                .authorizeRequests()
+                .anyRequest().permitAll();
+        http.csrf().disable();
+        http.headers().frameOptions().disable();
```
Comment (varunmehta) [inline src/main/java/org/glenrockindiancommunity/WebSecurityConfig.java]: Why is this disabled ?

File: src/main/java/org/glenrockindiancommunity/WebSecurityConfig.java [hunk 2]
```diff
@@ -11,33 +12,31 @@
...
+    }
 
-}
+    protected void configure(AuthenticationManagerBuilder auth)
+            throws Exception {
+        auth.inMemoryAuthentication().withUser("admin1").password("secret1")
```
Comment (varunmehta) [inline src/main/java/org/glenrockindiancommunity/WebSecurityConfig.java]: Check what ROLES does the default "user" get

File: src/main/java/org/glenrockindiancommunity/WebSecurityConfig.java (hunk 1 above)
Comment (kunalkhosla) [inline src/main/java/org/glenrockindiancommunity/WebSecurityConfig.java]: Becasue of : https://spring.io/blog/2013/08/23/spring-security-3-2-0-rc1-highlights-security-headers/#x-frame-options

File: src/main/java/org/glenrockindiancommunity/WebSecurityConfig.java (hunk 2 above)
Comment (kunalkhosla) [inline src/main/java/org/glenrockindiancommunity/WebSecurityConfig.java]: There is no default role called user. There is a line below line # 39. Which has the ADMIN role assigned to this admin1 user.

File: src/main/java/org/glenrockindiancommunity/controller/HomeController.java [hunk 3]
```diff
@@ -18,4 +27,36 @@ public String redirectToIndex(Model model) {
...
+
+    if (principal instanceof UserDetails) {
+      userName = ((UserDetails) principal).getUsername();
+    } else {
+      userName = principal.toString();
+    }
+    return userName;
+  }
+
 }
```
Comment (varunmehta) [inline src/main/java/org/glenrockindiancommunity/controller/HomeController.java]: With spring-boot, you don't need to do so much config. This is a mix of old and new style.

**Pull request closed without merging**